sdk/src/DocuSign.eSign/Client/ApiClient.cs

```diff
@@ -0,0 +1,1235 @@
+        /// <param name="userId"></param>
+        /// <param name="oauthBasePath"></param>
+        /// <param name="privateKeyFilename"></param>
+        /// <param name="expiresInHours"></param>
+        /// <param name="scopes"></param>
+        [Obsolete("This method is deprecated. Please use 'RequestJWTUserToken' instead.", false)]
+        public void ConfigureJwtAuthorizationFlow(string clientId, string userId, string oauthBasePath, string privateKeyFilename, int expiresInHours, List<string> scopes = null)
+        {
+            if (!string.IsNullOrEmpty(privateKeyFilename))
+            {
+                byte[] privateKeyBytes = File.ReadAllBytes(privateKeyFilename);
+                this.RequestJWTUserToken(clientId, userId, oauthBasePath, privateKeyBytes, expiresInHours, scopes);
+            }
+            else
+            {
+                throw new ApiException(400, "Private key not supplied or is invalid!");
+            }
+        }
+
+        /// <summary>
+        /// ConfigureJwtAuthorizationFlowByKey which performs JWT authentication using the private key. 
+        /// </summary>
+        /// <param name="clientId"></param>
+        /// <param name="userId"></param>
+        /// <param name="oauthBasePath"></param>
+        /// <param name="privateKey"></param>
+        /// <param name="expiresInHours"></param>
+        /// <param name="scopes"></param>
+        /// <returns>If Successful, returns the OAuthToken object model which consist of an access token and expiration time.</returns>
+        [Obsolete("This method is deprecated. Please use 'RequestJWTUserToken' instead.", false)]
+        public OAuth.OAuthToken ConfigureJwtAuthorizationFlowByKey(string clientId, string userId, string oauthBasePath, string privateKey, int expiresInHours, List<string> scopes = null)
+        {
+            if (!string.IsNullOrEmpty(privateKey))
+            {
+                byte[] privateKeyBytes = Encoding.UTF8.GetBytes(privateKey);
+                return this.RequestJWTUserToken(clientId, userId, oauthBasePath, privateKeyBytes, expiresInHours, scopes);
+            }
+            throw new ApiException(400, "Private key not supplied or is invalid!");
+        }
+
+        /// <summary>
+        /// Creates an RSA Key from the given PEM key.
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns>RSACryptoServiceProvider using the "key"</returns>
+        private static RSA CreateRSAKeyFromPem(string key)
+        {
+            TextReader reader = new StringReader(key);
+            PemReader pemReader = new PemReader(reader);
+
+            object result = pemReader.ReadObject();
+
+#if NETSTANDARD2_0
+           var provider = new RSACryptoServiceProvider();
+#else
+            var cspParameters = new CspParameters
+            {
+                Flags = CspProviderFlags.UseMachineKeyStore,
+            };
+
+            var provider = new RSACryptoServiceProvider(cspParameters);
+#endif
```

Comment:
Since the key isn't named, the `UseMachineKeyStore` shouldn't have any discernible effect; meaning this whole thing could be either

```C#
var provider = new RSACryptoServiceProvider();
```

or, even better, avoid directly using RSACryptoServiceProvider (on .NET Core this change means you gain support for RSA-PSS signatures and RSA-OAEP-SHA2 encryption; on .NET Framework it would still be RSACryptoServiceProvider by default).

```C#
RSA provider = RSA.Create();
```

An alternative, if the UseMachineKeyStore is having meaning that I'm spacing on right now for unnamed keys:

```C#
RSA provider;

if (RuntimeEnvironment.IsOSPlatform(OSPlatform.Windows))
{
    var cspParameters = new CspParameters
    {
        Flags = CspProviderFlags.UseMachineKeyStore,
    };

    provider = new RSACryptoServiceProvider(cspParameters);
}
else
{
    provider = new RSACryptoServiceProvider();
}
```

(System.Runtime.InteropServices namespace)

Reply:
@bartonjs i have a test prerelease here https://github.com/docusign/docusign-csharp-client/pull/232